tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,420 @@
+"""Unit-Test Case for otx.cli.builder.builder."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+
+import mmcv
+import pytest
+from mmcv.utils import Registry
+from torch import nn
+
+from otx.cli.builder.builder import (
+    Builder,
+    get_backbone_out_channels,
+    update_backbone_args,
+    update_channels,
+)
+from otx.cli.utils.importing import get_otx_root_path
+from otx.mpa.utils.config_utils import MPAConfig
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+
+
+class TestOTXCLIBuilder:
+    """Check Builder's function is working well.
+
+    1. Check "Builder.build_task_config" function that create otx-workspace is working well.
+    <Steps>
+        1. Create Classification custom workspace
+        2. Raising Error of building workspace with already created path
+        3. Update hparam.yaml with train_type="selfsl"
+        4. Raising ValueError with wrong train_type
+        5. Build workspace with model_type argments
+        6. Raise ValueError when build workspace with wrong model_type argments
+
+    2. Check "Builder.build_backbone_config" function that generate backbone configuration file is working well
+    <Steps>
+        1. Generate backbone config file (mmcls.MMOVBackbone)
+        2. Raise ValueError with wrong output_path
+
+    3. Check "Builder.build_model_config" function that update model config with new backbone is working well
+    <Steps>
+        1. Update model config with mmcls.ResNet backbone (default model.backbone: otx.OTXEfficientNet)
+        2. Raise ValueError with wrong model_config_path
+        3. Raise ValueError with wrong backbone_config_path
+        4. Update model config without backbone's out_indices
+        5. Update model config with backbone's pretrained path
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, tmp_dir_path):
```

Comment:
Is `tmp_dir_path` still a `Path` object? If not can we use Path object since it is much more convenient than `str` and `os`

Reply:
I added a type hint with `Union[Path, str]`.